index.bs

```diff
@@ -4109,165 +4301,1226 @@ To <dfn>get the response data</dfn> given |response|:
+       with [=error code=] [=invalid argument=].
+
+    1. If |codepoint| is U+005C (\):
+
+        1. Set |is escaped character| to true.
+
+        1. Continue.
+
+  1. Append |codepoint| to result.
+
+  1. Set |is escaped character| to false.
+
+1. Return [=success=] with data |result|.
+
+</div>
+
+<div algorithm>
+To <dfn>parse URL pattern</dfn>, given |pattern|:
+
+1. Let |has protocol| be true.
+
+1. Let |has hostname| be true.
+
+1. If |pattern| matches the <code>network.UrlPatternPattern</code> production:
+
+  1. Let |pattern url| be the empty string.
+
+    1. If |pattern| [=map/contains=] "<code>protocol</code>":
+
+      1. Let |protocol| be the result of [=trying=] to [=unescape
+         URL Pattern=] with |pattern|["<code>protocol</code>"].
+
+      1. For each |codepoint| in |protocol|:
+
+      1. If |codepoint| is not [=ASCII alphanumeric=] and «U+002B (+), U+002D
+         (-), U+002E (.)» does not [=set/contains|contain=] |codepoint|:
+
+        1. Return [=error=] with [=error code=] invalid argument.
+
+      1. Append |protocol| to |pattern url|.
+
+    1. Otherwise:
+
+      1. Set |has protocol| to false.
+
+      1. Append "<code>http</code>" to |pattern url|.
+
+    1. Let |scheme| be [=ASCII lowercase=] with |pattern url|.
+
+    1. Append "<code>:</code>" to |pattern url|.
+
+    1. If |scheme| [=is special=], append "<code>//</code>" to |pattern url|.
+
+    1. If |pattern| [=map/contains=] "<code>hostname</code>":
+
+      1. If |scheme| is "<code>file</code>" return [=error=] with [=error code=]
+         [=invalid argument=].
+
+      1. Let |hostname| be the result of [=trying=] to [=unescape
+         URL Pattern=] with |pattern|["<code>hostname</code>"].
+
+      1. Let |inside brackets| be false.
+
+      1. For each |codepoint| in |hostname|:
+
+        1. If «U+003A (:), U+002F (/), U+003F (?), U+0023 (#)» [=set/contains=]
+           |codepoint|:
+
+          1. Return [=error=] with [=error code=] [=invalid argument=].
+
+        1. If |inside brackets| is false and |codepoint| is U+003A (:):
+
+          1. Return [=error=] with [=error code=] [=invalid argument=].
+
+        1. If |codepoint| is U+005B ([), set |inside brackets| to true.
+
+        1. If |codepoint| is U+005D (]), set |inside brackets| to false.
+
+      1. Append |hostname| to |pattern url|.
+
+    1. Otherwise:
+
+      1. If |scheme| is not "<code>file</code>", append "<code>placeholder</code>" to
+         |pattern url|.
+
+      1. Set |has hostname| to false.
+
+    1. If |pattern| [=map/contains=] "<code>port</code>":
+
+      1. Let |port| be the result of [=trying=] to [=unescape
+         URL Pattern=] with |pattern|["<code>port</code>"].
+
+      1. Append "<code>:</code>" to |url pattern|.
+
+      1. For each |codepoint| in |port|:
+
+        1. If |codepoint| is not an [=ASCII digit=]:
+
+          1. Return [=error=] with [=error code=] [=invalid argument=].
+
+        1. Append |port| to |url pattern|.
```

Comment:
```suggestion
      1. Append |port| to |url pattern|.
```

Indentation seems wrong here, should only be done after checking all codepoints, not for each one of them.